cmake/Dependencies.cmake

```diff
@@ -273,21 +273,8 @@ endif()
 # instead of the default implementation. To avoid confusion, add pthreadpool
 # subdirectory explicitly with EXCLUDE_FROM_ALL property prior to QNNPACK/NNPACK
 # does so, which will prevent it from installing the default pthreadpool library.
-if(INTERN_BUILD_MOBILE AND NOT BUILD_CAFFE2_MOBILE AND (USE_QNNPACK OR USE_NNPACK OR USE_XNNPACK))
```

Comment:
In addition to this change, I think you will need modify the list of static libraries to link at the following places:
https://github.com/pytorch/pytorch/blob/master/android/pytorch_android/CMakeLists.txt#L89
https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/binary_ios_upload.sh#L17
https://github.com/pytorch/pytorch/blob/master/scripts/xcode_build.rb#L54
https://github.com/pytorch/pytorch/blob/master/ios/TestApp/benchmark/setup.rb#L66

I hope we can consolidate all these to:
https://github.com/pytorch/pytorch/blob/master/cmake/TorchConfig.cmake.in#L78
E.g. use case: https://github.com/pytorch/pytorch/blob/master/test/mobile/op_deps/CMakeLists.txt#L27

But this should be a separate change.